### DISCUSSION TITLE
moving HR's from one Kustomization to another

### DISCUSSION THREAD

**Discussion by jack-evans** [fluxcd/flux2#2318] [Q&A] [answered]
Currently, we use flux (v1) and are in the process of moving to flux2.

in flux1 we had a format of our repo:
```
cluster-configuration/
  charts/
    ...
  releases-cp/
    uservice-A.yaml
    uservice-B.yaml
    istio.yaml
    ...
```
(cp for control-plane) 

The aim is produce the exact same in flux2, which is pretty straightforward (kustomization that points to releases-cp) and we have that working in a number of environments... However, now I want to look towards having things install in an order. For example istio, needs to be installed first before Installing anything else so to me it makes sense to have something like:

```
cluster-configuration/
  charts/
    ...
  releases-infra/
    cp/
      istio.yaml
  releases-cp/
    uservice-A.yaml
    uservice-B.yaml
    ...
```

and then use kustomizations `dependsOn` to wait for a kustomization pointing to `./releases-infra/cp` before allowing the kustomization pointing to `releases-cp` to reconcile.... 

How easy is it to move a Helm release from one kustomization to another?

> only thing I can think potentially would happen is that the first kustomization deletes it after the second kustomization reconciles successfully...

**Comment by stefanprodan** [comment, marked as answer]
To move a resource from one Kustomization to another, you can suspend the original Kustomization, move the resource in Git, reconcile the destination Kustomization then resume the original one.

**Reply by jack-evans** [reply]
ahh awesome! 😍 So I could create the new directories in advance, with kustomizations pointing at them (but reconciling nothing). suspend the old ones, move the resources in git (either gradually or in one jump) which would then reconcile it under the new one.

**Reply by stefanprodan** [reply]
Yes, we have safe guards in place for GC, kustomize-controller will notice that a resource that it should delete is owned by another Kustomization and it will skip it. Just make sure to suspend the original ones and reconcile the new ones before resuming. Another option is to disable GC on the old ones with `prune: false`.
